tests/functional/cli/test_factory.py

```diff
@@ -97,3 +97,18 @@ def test_filename_and_lineno_included_in_syntax_error(clifactory):
     message = str(excinfo.value)
     assert 'app.py' in message
     assert 'line 1' in message
+
+
+def test_can_import_vendor_package(clifactory):
+    # Tests that vendor packages can be imported during config loading.
+    vendedlib_dir = os.path.join(clifactory.project_dir, 'vendor', 'vendedlib')
+    os.makedirs(vendedlib_dir)
+    open(os.path.join(vendedlib_dir, '__init__.py'), 'a').close()
+    open(os.path.join(vendedlib_dir, 'submodule.py'), 'a').close()
+    app_py = os.path.join(clifactory.project_dir, 'app.py')
+    with open(app_py, 'r+') as f:
+        data = f.read()
+        f.seek(0, 0)
+        f.write('from vendedlib import submodule\n%s' % data)
+    config = clifactory.create_config_obj()
```

Comment:
So after looking at this again, I'd still prefer the test to be explicit.  This is implicitly relying on the fact that if we can't import the module then this code will propagate some exception in `create_config_obj`.  I'd rather be more explicit, ensure that not only do we not raise an exception but we are in fact able to access the thing we're importing.  As an example:

```
# submodule.py:

SOME_VALUE = 'from-submodule'

# app.py

from vendedlib import submodule
app = Chalice(app_name='foo')
app.from_submodule = submodule.SOME_VALUE

# test
assert app.from_submodule == 'from-submodule'
```

Or something along those lines.  It's just more explicit about successfully importing the vendored submodule